src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.SparkProgramGroup;
+import org.broadinstitute.hellbender.engine.spark.GATKSparkTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadConstants;
+import scala.Tuple2;
+
+import java.io.*;
+import java.util.*;
+
+@CommandLineProgramProperties(summary="Produce FASTQs of breakpoint candidates",
+        oneLineSummary="Produce mini-FASTQs of reads sharing kmers with putative breakpoints for local assembly",
+        programGroup = SparkProgramGroup.class)
+public class GatherSplitReadsSpark extends GATKSparkTool
+{
+    private static final long serialVersionUID = 1L;
+    private static final int KLEN = 63;
+
+    @Argument(doc = "dir for output FASTQs", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, optional = false)
+    private String outputDir;
+
+    // This is a path to a file of kmers that appear too frequently in the reference to be usable as probes to localize
+    // reads.  We don't calculate it here, because it depends only on the reference.
+    // The program FindBadGenomicKmers can produce such a list for you.
+    @Argument(doc = "ubiquitous kmer list", shortName = "KS", fullName = "kmerKillSet", optional = false)
+    private String kmerKillSet;
+
+    @Override
+    public boolean requiresReads()
+    {
+        return true;
+    }
+
+    @Override
+    protected void runTool( final JavaSparkContext ctx )
+    {
+        final SAMFileHeader header = getHeaderForReads();
+        if ( header.getSortOrder() != SAMFileHeader.SortOrder.coordinate )
+            throw new GATKException("The BAM must be coordinate sorted.");
+
+        // Process the input data to produce a map of Kmer -> List of breakpoint IDs.
+        // (A breakpoint ID is just an arbitrary identifier for the putative breakpoint.)
+        // See the doc for findClusters to see a description of how this big Spark pipeline works.
+        final Broadcast<Map<Kmer,List<Long>>> broadcastMap = ctx.broadcast(findClusters(ctx));
+
+        // Process the input again, this time pulling all reads that contain kmers associated with a given breakpoint,
+        // and writing those reads into a separate FASTQ for each breakpoint.
+        getGoodReads()
+            .mapPartitionsToPair(readItr -> new ReadPuller(readItr,broadcastMap.value()))
+            .groupByKey()
+            .foreach(tupl -> writeFastq(tupl._1,tupl._2.iterator(),outputDir));
+    }
+
+    public JavaRDD<GATKRead> getGoodReads() {
```

Comment:
The value added is improved consistency across tools. GATK has traditionally had an engine-level argument to activate and de-activate read filters that works for all tools transparently, and this is one of the design requirements of GATK4 as well. If every tool is doing their own thing when it comes to read filters this task will be made difficult/impossible.